In [7]:
import pandas as pd
import numpy as np
import seaborn as sns

# Load penguins
penguins = sns.load_dataset("penguins")

# Quick Look 
penguins.head()

# How many rows and columns
penguins.shape

# What are the column name
penguins.columns

# Data counts and non-null counts (quick overview)
penguins.info()
penguins.isna().sum()

# Basic stats for numeric columns
penguins.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

### IQR = Interquartile range
- 25% Lower quartile (Q1), 25% of values fall below this number
- 50% Median middle value
- 75% Upper quartile (Q3), 75% of values rise above this number
- IQR = Q3 - Q1 -> the spread of the 'middle 50%' of data

#### Outlier are usually flagged using the 1.5 * IQR rule:
- Lower bound = Q1 - (1.5*IQR)
- Upper bound = Q3 + (1.5*IQR)
- Anything outside these are considered outliers

#### Example: body_mass_g
- Q1 = 3550, Q3 = 4750
- IQR = (Q3 - Q1) = 1200
- What is 1.5 * 1200? 1800
- Lower bound = 3550 - 1800 = 1750
- Higher bound = 4750 + 1800 = 6550